modules/message-lane/src/lib.rs

```diff
@@ -336,12 +336,13 @@ decl_module! {
 				// reward relayers that have delivered messages
 				// this loop is bounded by `T::MaxUnconfirmedMessagesAtInboundLane` on the bridged chain
 				for (nonce_low, nonce_high, relayer) in lane_data.relayers {
-					for nonce in nonce_low..nonce_high + 1 {
-						// we may receive the same confirmation multiple times
-						if nonce_high < received_range.0 {
-							continue;
-						}
-
+					let nonce_begin = sp_std::cmp::max(nonce_low, received_range.0);
+					let nonce_end = sp_std::cmp::min(nonce_high, received_range.1);
+					// don't proceed if this messages entry is ahead of received range.
+					if nonce_low > nonce_end {
```

Comment:
This check is actually redundant now :) It is done by the for loop. Let's remove it probably?

Reply:
You're right, bc of the range.